Docker-in-docker feature support  for fedora

sireeshajonnalagadda opened the pull request:
Feature Name

- Docker-in-docker

Description of Changes

- Extending support of Docker-in-docker feature for other linux distributions (Fedora)

- As per the user request https://github.com/devcontainers/features/issues/1235

Changelog

- Modified the install.sh to work the expected installation for Fedora

- Wrote test case scenarios to meet the required checks for the Docker-in-docker

Checklist

- [ ☑️ ]  Checked that applied changes work as expected

Kaniska244 commented on src/docker-in-docker/install.sh:
```diff
@@ -247,7 +279,7 @@ fi
 # Refresh apt lists
 apt-get update
-
+fi
```
Please correct the indentation for better readability.

Kaniska244 commented on src/docker-in-docker/install.sh:
```diff
@@ -62,19 +62,37 @@ fi
+
 }
+# Install dependencies for fedora
+if ! command -v git wget which&> /dev/null; then
```
Just out of curiosity, how is this condition only passing for fedora & not for debian or ubuntu?

Kaniska244 commented on src/docker-in-docker/install.sh:
```diff
@@ -62,19 +62,37 @@ fi
         apt-get update -y
+        apt-get install -y gnupg curl
+        apt-get -y install --no-install-recommends "$@"
+       fi
```
These installations should not be done in this `apt_get_update()` function. Would it possible to move them below as part of code in this [code](https://github.com/devcontainers/features/pull/1328/files#diff-a72a4e49c44e54f09feb84b7767d355d68d34fbee561d8f96a99d12d2488b9c5L215), if the package isn't included there already.

sireeshajonnalagadda commented:
Hi @kaniska, 
Made the changes according to the review comments, please review and let me know in case of any further concerns/clarity.

Kaniska244 commented on src/docker-in-docker/install.sh:
```diff
@@ -293,33 +327,93 @@ if [ "${USE_MOBY}" = "true" ]; then
+        echo "Installing Moby packages..."
+        sudo dnf install -y moby-engine moby-cli moby-buildx --skip-unavailable
+        
+    else
```
Hi @sireeshajonnalagadda ,

When does the code inside this `else` condition get executed ? I see [here](https://github.com/devcontainers/features/pull/1328/files#diff-a72a4e49c44e54f09feb84b7767d355d68d34fbee561d8f96a99d12d2488b9c5R404) this `install_docker_or_moby` function called only when `USE_MOBY` flag is set to `true`. Please let me know if I am missing something.

Kaniska244 commented on test/docker-in-docker/scenarios.json:
```diff
@@ -166,5 +166,16 @@
+        "features": {
+            "docker-in-docker": {
+                "iptables": "true",
+                "moby": "true",
```
Hi @sireeshajonnalagadda ,

We should have another test case also with `"moby": "false"` for fedora

sireeshajonnalagadda commented:
HI @Kaniska244, 
Addressed the review comments and made few changes that works for both ubuntu and fedora, please review the latest changes and let me know in case of any Queries!
